packages/firestore/src/local/indexeddb_mutation_queue.ts

```diff
@@ -103,38 +89,45 @@ export class IndexedDbMutationQueue implements MutationQueue {
   }
 
   start(transaction: PersistenceTransaction): PersistencePromise<void> {
+    let nextBatchId;
+
     return IndexedDbMutationQueue.loadNextBatchIdFromDb(transaction)
-      .next(nextBatchId => {
-        this.nextBatchId = nextBatchId;
+      .next(batchId => {
+        nextBatchId = batchId;
         return mutationQueuesStore(transaction).get(this.userId);
       })
       .next((metadata: DbMutationQueue | null) => {
         if (!metadata) {
           metadata = new DbMutationQueue(
             this.userId,
             BATCHID_UNKNOWN,
-            /*lastStreamToken=*/ ''
+            /*lastStreamToken=*/ '',
+            BATCHID_UNKNOWN
           );
         }
-        this.metadata = metadata;
+
+        // TODO(multitab): This should only run when no other client is
+        // active, as batch IDs may otherwise go back in time in clients that
+        // have already processed mutations.
 
         // On restart, nextBatchId may end up lower than
         // lastAcknowledgedBatchId since it's computed from the queue
```

Comment:
Is this true any more? I thought your change was preventing reuse of batch IDs? And so we shouldn't ever reset lastAcknowledgedBatchId...

Reply:
Ahh...  Yes, I missed that.  I'm inclined to get rid of all this logic and just use auto-incrementing IDs that never recycle (either by leveraging IndexedDB's autoincrement feature or manually).